src/lib.rs

```diff
@@ -52,18 +68,45 @@ impl<T: 'static + Send + Sync> Default for DefaultRaycastingPlugin<T> {
 pub enum RaycastSystem {
     BuildRays,
     UpdateRaycast,
+    #[cfg(feature = "debug")]
+    RemoveDebugCursors,
+    #[cfg(feature = "debug")]
     UpdateDebugCursor,
 }
 
 /// Global plugin state used to enable or disable all ray casting for a given type T.
 pub struct PluginState<T> {
     pub enabled: bool,
+    #[cfg(feature = "debug")]
+    pub debug: bool,
     _marker: PhantomData<T>,
 }
+
+impl<T> PluginState<T> {
+    #[cfg(feature = "debug")]
+    pub fn new(enabled: bool, debug: bool) -> Self {
```

Comment:
I'd like to replace the booleans here with some enums to make this more explicit. Something like 

```rust
pub fn new(state: State, debug: Debug) -> Self {
```

where

```rust
pub enum State {
    Enabled,
    Disabled,
}

pub enum Debug {
    Enabled,
    Disabled,
}
```

That way, use of `new()` looks like this in practice:

```rust
commands.insert_resource(PluginState::<MyRaycastSet>::new(State::Enabled, Debug::Disabled));
```

Reply:
Good idea, I will implement it.
A thing to be aware of though is, that moving the enabled state from `bool` to `enum` is a braking change. However I already bumped the version in this PR to `0.3.0` so this would be reflected.

Reply:
Yep! I think the version bump is sufficient. Thanks!